src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -89,53 +89,144 @@
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   private int a;
+ *   private int b;
+ *   private int c;
+ *
+ *   public Test(int a) {
+ *     // overlapping by constructor argument
+ *     this.a = a;       // OK, this keyword used
+ *     b = 0;            // OK, no overlap
+ *     foo(arg);         // OK, no overlap
```

Comment:
`foo(arg)` what is `arg` here?

Reply:
I have just used `arg` to indicate that it could be any method with or without **argument**. Let me know if I should remove it, if it seems unnecessary.

Reply:
I mean that there is no such field nor parameter, so it is confusing

Reply:
Okay, I shall use an integer argument instead, so that there is no need of `arg`

Reply:
now it looks even more confusing as before.
`foo(5) // OK, no overlap` where overlap can be? It is just invocation of method, it is not clear how it is related with this check and its purpose

Reply:
That is true, there can be no overlap for a method. The only case where it would raise violation is when `validateOnlyOverlapping` is `false`, and `checkMethods` is `true`.
I will change `// OK, no overlap` to just `// OK`